mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala

```diff
@@ -38,16 +38,14 @@ import org.apache.spark.rdd.RDD
  *                            Since 3.4.0, it supports ndcg evaluation with relevance value.
  */
 @Since("1.2.0")
-class RankingMetrics[T: ClassTag] @Since("3.4.0") (
-    predictionAndLabels: RDD[(Array[T], Array[T], Array[Double])])
+class RankingMetrics[T: ClassTag] @Since("1.2.0") (predictionAndLabels: RDD[_ <: Product])
     extends Logging
     with Serializable {
 
-  @Since("1.2.0")
-  def this(predictionAndLabelsWithoutRelevance: => RDD[(Array[T], Array[T])]) = {
-    this(predictionAndLabelsWithoutRelevance.map {
-      case (pred, lab) => (pred, lab, Array.empty[Double])
-    })
+  private val rdd = predictionAndLabels.map {
```

Comment:
Maybe we can rename this and the constructor arg; the constructor arg may also have relevance; this does not

Reply:
We may want not to change the name because 
- `MulticlassMetrics` also has the arguments whose name is `predictionAndLabels`.
https://github.com/apache/spark/blob/b588d070ebc234280af730c8f9915e8859b1886e/mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala#L28-L35
- We change the interface of the constructor to be able to support more inputs. Specific names like `predictionAndLabelsWithOptionalRelevance` may go against the goal.

Reply:
OK that's reasonable

Reply:
why discarding `rel` here?

If `rdd` is a `RDD[Array[T], Array[T], Array[Double]]`, then in the `ndcgAt`, we can simply check whether `rel` is empty?

also, maybe we can rename `rdd` to a more meaningful name

Reply:
WDYT @uchiiii ?

Reply:
How about `fullRDD`? 
IMO, Names like `predictionsAndLabelsAndRelevances` are too long.

Reply:
maybe `predictionsLabelsRelevances`?